base/compiler/ssair/inlining.jl

```diff
@@ -843,7 +859,7 @@ function resolve_todo(todo::InliningTodo, state::InliningState, flag::UInt8)
         if code isa CodeInstance
             if use_const_api(code)
                 # in this case function can be inlined to a constant
-                et !== nothing && add_edge!(et, invokesig, mi)
+                add_inlining_backedge!(et, mi)
```

Comment:
@Keno why do we need to account for `mi`-backedge for this constant result case? The `mi` is retrieved here https://github.com/JuliaLang/julia/blob/59daacb53ea347a9aa30aaffffe61d99b254cbd2/base/compiler/ssair/inlining.jl#L954
and so the abstract interpretation should account for it already.

Reply:
We were probably just being conservative here. We didn't used to be particularly careful about whether information was being computed during abstract interpretation or in the optimizer.

Reply:
I see. It seems a bit inconsistent that we doesn't account for `mi`-backedge for [semi-]concrete evaluated case, e.g.: https://github.com/JuliaLang/julia/blob/59daacb53ea347a9aa30aaffffe61d99b254cbd2/base/compiler/ssair/inlining.jl#L1464-L1474

Reply:
Well, come up with something that has a constant return that gets inlined here, add a test for it and then delete the backedge here. If things pass (and go through PkgEval), I'm comfortable to remove it.